llvm/include/llvm/IR/DebugInfoMetadata.h

```diff
@@ -3073,6 +3073,109 @@ template <> struct DenseMapInfo<DIExpression::FragmentInfo> {
   static bool isEqual(const FragInfo &A, const FragInfo &B) { return A == B; }
 };
 
+class DIExpressionOptimizer {
```

Comment:
Can you add a Doxygen comment for the entire class with just a sentence on what this does?